buildSrc/src/main/groovy/org/gradle/binarycompatibility/rules/BinaryBreakingChangesRule.java

```diff
@@ -24,21 +25,31 @@
 
 public class BinaryBreakingChangesRule extends AbstractGradleViolationRule {
 
-    public BinaryBreakingChangesRule(Map<String, String> acceptedViolations) {
-        super(acceptedViolations);
+    public BinaryBreakingChangesRule(Map<String, String> acceptedApiChanges) {
+        super(acceptedApiChanges);
     }
 
+    private JApiClass currentClass;
+
     @Override
     @SuppressWarnings("unchecked")
     public Violation maybeViolation(final JApiCompatibility member) {
         if (!member.isBinaryCompatible()) {
+            if (member instanceof JApiClass) {
+                currentClass = (JApiClass) member;
+                if (member.getCompatibilityChanges().isEmpty()) {
+                    // A member of the class breaks binary compatibility.
+                    // That will be handled when the member is passed to `maybeViolation`.
+                    return null;
+                }
+            }
             if (member instanceof JApiHasAnnotations) {
                 if (isIncubating((JApiHasAnnotations) member)) {
                     return Violation.accept(member, "Removed member was incubating");
                 }
             }
 
-            return acceptOrReject(member, Violation.notBinaryCompatible(member));
+            return acceptOrReject(currentClass, member, Violation.notBinaryCompatible(member));
```

Comment:
I think nested classes are treated completely separate. That means that first the whole class with its members (nested classes are not considered as members) is passed to the rule, and then the nested class with all its member is passed. Still, I need to confirm that. I should probably talk to @melix to adjust the API and include the parent class as an argument to `maybeViolation` or provide it via the context.